### PR TITLE
Add Integration Test

### PR DESCRIPTION
**Pull request by Sophylax**
This adds an integration test to check different parts of the trapper are working together properly. Previously we were only testing individual components, but not whether we can execute a training experiment and confirm that it results in a model that is trained.

The new test runs a training experiment on a small model from scratch. It then saves and loads the model. Finally, it runs the loaded model on several inputs to see if the model behaves in an expected manner. The test is mostly to see if this process can be completed without any errors.

### CODE REVIEW

File: test_fixtures/pipelines/experiment.jsonnet [hunk 1]
```diff
@@ -0,0 +1,66 @@
+local checkpoint_dir = std.extVar("CHECKPOINT_PATH");
```
Comment (cemilcengiz) [inline test_fixtures/pipelines/experiment.jsonnet:1]: Shall we rename this file to something like `pipeline_integration_experiment.json` if it is only used in `test_integration.py`?

File: tests/pipelines/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    checkpoint_dir = os.path.join(experiment_dir, "checkpoints")
+    output_dir = os.path.join(experiment_dir, "outputs")
+
+    ext_vars = {
+        # Used to feed the jsonnet config file with file paths
+        "OUTPUT_PATH": output_dir,
+        "CHECKPOINT_PATH": checkpoint_dir
+    }
+
+    result = run_experiment(
```
Comment (cemilcengiz) [inline tests/pipelines/test_integration.py:81]: This line crashes with `FileNotFoundError: Couldn't find a dataset script at trapper/tests/test_fixtures/hf_datasets/squad_qa_test_fixture/squad_qa_test_fixture.py or any data file in the same directory.
`

File: tests/pipelines/test_integration.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+from trapper.pipelines.pipeline import create_pipeline_from_checkpoint
+from trapper.pipelines.question_answering_pipeline import (
+    SquadQuestionAnsweringPipeline,
+)
+from trapper.training.train import run_experiment
+
+PIPELINE_FIXTURES = FIXTURES_ROOT / "pipelines"
+
+@pytest.fixture(scope="module")
+def integration_training_result():
```
[the remark this answers is not in the document]
Comment (Sophylax) [reply tests/pipelines/test_integration.py]: Done.

File: tests/pipelines/test_integration.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    qa_pipeline = create_pipeline_from_checkpoint(
+        checkpoint_path=PRETRAINED_MODEL_PATH,
+        experiment_config_path=EXPERIMENT_CONFIG,
+        task="squad-question-answering",
+    )
+
+    actual_single_inference = qa_pipeline(integration_single_inference_input)
+
+    diff = DeepDiff(actual_single_inference, integration_expected_single_inference, significant_digits=3)
+    assert diff == {}, f"Actual and Desired Dicts are not Almost Equal:\n {pprint.pformat(diff, indent=2)}"
```
[the remark this answers is not in the document]
Comment (Sophylax) [reply tests/pipelines/test_integration.py]: Simplified the assertions.

File: test_fixtures/pipelines/experiment.jsonnet (hunk 1 above)
Comment (Sophylax) [reply test_fixtures/pipelines/experiment.jsonnet]: Done.

File: tests/pipelines/test_integration.py (hunk 2 above)
Comment (Sophylax) [reply tests/pipelines/test_integration.py:80]: This looks like the test is not running in the project root as the active directory. Due to experiment config specifying a relative path, that might be the cause of that problem. I specified the path relative to the project root due to where we are running the test in GitHub test action.

**Comment by cemilcengiz** [comment]
LGTM!

**Review by cemilcengiz — APPROVED** [review]

**Pull request merged**